src/babel/api/register/node.js

```diff
@@ -160,7 +161,7 @@ hookExtensions(util.canCompile.EXTENSIONS);
 
 export default function (opts = {}) {
   if (opts.only != null) only = util.arrayify(opts.only, util.regexify);
-  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify);
+  if (opts.ignore != null) ignore = util.arrayify(opts.ignore, util.regexify, isFunction);
```

Comment:
I see your point and agree with it too, but only downside is the function invocation becomes less I terse even when mapping function and filtration (exclusion) function are previously defined. 
Since its an opinion with upsides and downsides in both, I can change it if you feel strongly against it.